### PR TITLE
Adding node bootstrapping into openshift_node.

### PR DESCRIPTION
**Pull request by kwoodson**
The requirement for creating bootstrapping was to share the same openshift_node role so that we have a single location for configuring openshift_node.

I have refactored openshift_node to allow us to share pieces of the settings that are needed by both the bootstrap and non bootstrap versions.

I could not call into openshift_version because of some master requirements.

There are basically two paths through openshift_node now.  bootstrap and normal (non bootstrap).  bootstrap and normal share a bunch of tasks and settings.  bootstrap.yml has the settings needed so that the node will start up and attempt to bootstrap itself to the cluster.  The normal path should have not changed.
- The shared code inside of openshift_node was left inside of tasks/main.yml
- The bootstrap code is placed inside of bootstrap.yml
- The pieces to install the packages were placed inside of install.yml
- The normal path of config was placed inside of config.yml

### CODE REVIEW

**Comment by kwoodson** [comment]
aos-ci-test

**Comment by kwoodson** [comment]
aos-ci-test

File: playbooks/aws/README.md [hunk 1]
```diff
@@ -188,12 +157,46 @@ openshift_hosted_registry_wait=False
 
 There are more examples of cluster inventory settings [`here`](../../inventory/byo/).
 
+In order to create the bootstrapable AMI we need to create an openshift-ansible inventory file.  This file enables us to create the AMI using the openshift-ansible node roles.
+
+
+#### Step 2
+
+Once the vars.yml file has been updated with the correct settings for the desired AWS account then we are ready to build an AMI.
```
Comment (michaelgugino) [inline playbooks/aws/README.md]: What vars.yml file?  If you're referring the users to make in-tree changes, I don't think this is a good approach.

File: roles/openshift_node/defaults/main.yml [hunk 2]
```diff
@@ -1,6 +1,62 @@
...
+openshift_image_tag: ''
+
+openshift_node_ami_prep_packages:
+- "{{ openshift_service_type }}-master"
+- "{{ openshift_service_type }}-node"
+- "{{ openshift_service_type }}-docker-excluder"
+- "{{ openshift_service_type }}-sdn-ovs"
+- ansible
+- openvswitch
+- docker
```
Comment (michaelgugino) [inline roles/openshift_node/defaults/main.yml]: docker?  Don't we have a docker role?  I don't understand the point of adding these packages to the node role, maybe some of them, but I think the majority of them would be covered by other roles?

File: roles/openshift_node/tasks/bootstrap.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- name: update the ExecStart to have bootstrap
+  lineinfile:
+    dest: "/usr/lib/systemd/system/{{ openshift_service_type }}-node.service"
+    line: "{% raw %}ExecStart=/usr/bin/openshift start node --bootstrap --kubeconfig=${KUBECONFIG} $OPTIONS{% endraw %}"
+    regexp: "^ExecStart=.*"
+
+- name: "systemctl enable {{ openshift_service_type }}-node"
+  systemd:
+    name: "{{ item }}"
+    enabled: no
```
Comment (michaelgugino) [inline roles/openshift_node/tasks/bootstrap.yml]: This looks wrong.

**Comment by michaelgugino** [comment]
Talked about future role improvements in IRC.  LGTM.

**Comment by kwoodson** [comment]
aos-ci-test

**Comment by kwoodson** [comment]
[test]

**Comment by kwoodson** [comment]
@sdodson, Can we retest this?  This is the failure and I don't think its related.  I couldn't find a relating flake.
```
fatal: [ocp-master]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_args": {
            "daemon_reload": false, 
            "enabled": false, 
            "masked": true, 
            "name": "origin-master", 
            "no_block": false, 
            "state": "stopped", 
            "user": false
        }
    }
}

MSG:

Could not find the requested service origin-master: host
...ignoring
```

The code for this PR doesn't muck with the openshift_master/tasks/clean_systemd_units.yml.

**Comment by dobbymoodge** [comment]
rekick [test] b/c restart

**Comment by kwoodson** [comment]
aos-ci-test

**Comment by kwoodson** [comment]
[merge]

**Pull request merged**